File: src/main/java/at/deckweiss/logging/LoggerFactory.java
package at.deckweiss.logging;

public class LoggerFactory {
    private LoggerFactory() {
    }

    public static Logger getLogger(final Class<?> clazz) {
        return new LoggerImpl(org.slf4j.LoggerFactory.getLogger(clazz));
    }

    public static Logger getLogger(String name) {
        return new LoggerImpl(org.slf4j.LoggerFactory.getLogger(name));
    }
}

File: src/main/kotlin/at/deckweiss/logging/LoggerFactory.kt
package at.deckweiss.logging

import org.slf4j.LoggerFactory

object LoggerFactory {
    fun getLogger(clazz: Class<*>?): Logger {
        return LoggerImpl(LoggerFactory.getLogger(clazz))
    }

    fun getLogger(name: String?): Logger {
        return LoggerImpl(LoggerFactory.getLogger(name))
    }
}